core/src/main/java/org/elasticsearch/action/support/single/custom/TransportSingleCustomOperationAction.java

```diff
@@ -123,7 +124,32 @@ private AsyncSingleAction(Request request, ActionListener<Response> listener) {
             if (blockException != null) {
                 throw blockException;
             }
-            this.shardsIt = shards(clusterState, internalRequest);
+            ShardsIterator originalShardsIt = shards(clusterState, internalRequest);
```

Comment:
but I'm +1 with dropping the custom 'preferring local shard' logic. So lets drop that then?